exporter/dynatraceexporter/serialize.go

```diff
@@ -0,0 +1,208 @@
+	return serialize(dm)
+}
+
+func serializeIntSum(name, prefix string, dims dimensions.NormalizedDimensionList, t pdata.AggregationTemporality, dp pdata.IntDataPoint) (string, error) {
+	var valueOpt dtMetric.MetricOption
+	if t == pdata.AggregationTemporalityDelta {
+		valueOpt = dtMetric.WithIntCounterValueDelta(dp.Value())
+	} else {
+		valueOpt = dtMetric.WithIntCounterValueTotal(dp.Value())
+	}
+
+	dm, err := dtMetric.NewMetric(
+		name,
+		dtMetric.WithPrefix(prefix),
+		dtMetric.WithDimensions(dims),
+		dtMetric.WithTimestamp(dp.Timestamp().AsTime()),
+		valueOpt,
+	)
+
+	if err != nil {
+		return "", err
+	}
+
+	return serialize(dm)
+}
+
+func serializeDoubleSum(name, prefix string, dims dimensions.NormalizedDimensionList, t pdata.AggregationTemporality, dp pdata.DoubleDataPoint) (string, error) {
+	var valueOpt dtMetric.MetricOption
+	if t == pdata.AggregationTemporalityDelta {
+		valueOpt = dtMetric.WithFloatCounterValueDelta(dp.Value())
+	} else {
+		valueOpt = dtMetric.WithFloatCounterValueTotal(dp.Value())
+	}
+
+	dm, err := dtMetric.NewMetric(
+		name,
+		dtMetric.WithPrefix(prefix),
+		dtMetric.WithDimensions(dims),
+		dtMetric.WithTimestamp(dp.Timestamp().AsTime()),
+		valueOpt,
+	)
+
+	if err != nil {
+		return "", err
+	}
+
+	return serialize(dm)
+}
+
+func histMinMax(bounds []float64, counts []uint64) (float64, float64, bool) {
+	// Because we do not know the actual min and max, we estimate them based on the min and max non-empty bucket
+	minIdx, maxIdx := -1, -1
+	for y := 0; y < len(counts); y++ {
+		if counts[y] > 0 {
+			if minIdx == -1 {
+				minIdx = y
+			}
+			maxIdx = y
+		}
+	}
+
+	if minIdx == -1 || maxIdx == -1 {
+		return 0, 0, false
+	}
+
+	var min, max float64
+
+	// Use lower bound for min unless it is the first bucket, then use upper
+	if minIdx == 0 {
+		min = bounds[minIdx]
+	} else {
+		min = bounds[minIdx-1]
+	}
+
+	// Use upper bound for max unless it is the last bucket, then use lower
+	if maxIdx == len(counts)-1 {
+		max = bounds[maxIdx-1]
+	} else {
+		max = bounds[maxIdx]
+	}
+
+	return min, max, true
+}
+
+func serializeIntHistogram(name, prefix string, dims dimensions.NormalizedDimensionList, t pdata.AggregationTemporality, dp pdata.IntHistogramDataPoint) (string, error) {
+	if t == pdata.AggregationTemporalityDelta {
+		return "", errors.New("unsupported histogram temporality")
+	}
+
+	min, max, nonEmpty := histMinMax(dp.ExplicitBounds(), dp.BucketCounts())
+
+	if !nonEmpty {
+		return "", nil
+	}
+
+	dm, err := dtMetric.NewMetric(
+		name,
+		dtMetric.WithPrefix(prefix),
+		dtMetric.WithDimensions(dims),
+		dtMetric.WithTimestamp(dp.Timestamp().AsTime()),
+		dtMetric.WithIntSummaryValue(
+			int64(min),
+			int64(max),
+			dp.Sum(),
+			int64(dp.Count()),
+		),
+	)
+
+	if err != nil {
+		return "", err
+	}
+
+	return serialize(dm)
+}
+
+func serializeHistogram(name, prefix string, dims dimensions.NormalizedDimensionList, t pdata.AggregationTemporality, dp pdata.HistogramDataPoint) (string, error) {
```

Comment:
```suggestion
func serializeFloatHistogram(name, prefix string, dims dimensions.NormalizedDimensionList, t pdata.AggregationTemporality, dp pdata.HistogramDataPoint) (string, error) {
```

Reply:
I decided to keep Double/Float out of the name because it is left off the name in the collector code. They only have `Histogram` and `IntHistogram`. If you feel strongly I can make this change.

Reply:
You are right, I even think `IntHistogram` is deprecated. Lets keep it like it is.

Reply:
It is deprecated and was removed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/4276